src/Admin/Views/Organizations/_OrganizationForm.cshtml

```diff
@@ -0,0 +1,253 @@
+@using Bit.SharedWeb.Utilities
+@model OrganizationEditModel
+
+<form method="post" id="edit-form">
+    @if (Model.Provider != null)
+    {
+        @Html.HiddenFor(m => m.Provider.Id)
+        @Html.HiddenFor(m => m.Provider.Type)
+    }
```

Comment:
Why do we need these hidden fields? Is it to make this data flow through when the form is submitted?

Reply:
MVC needs the fields to exist within the `form` so then the values can be sent to the `POST` action.

Reply:
Using hidden fields to sneak important data into a POST request feels a bit hacky. It works, but it makes me wonder whether there's a better way to solve this.

What if we moved the endpoint to the `Providers` controller and put it at (for example) `providers/createorganization/{providerId}`? A few arguments in favour of this:
* puts the providerId in the path rather than hidden fields. If you need any more info about that provider after submission, you can just query the db
* consistent with `providers/addexistingorganization/{providerId}` which is a similar flow
* it's a provider-specific flow: you can only ever create an organization via a provider, and it'll always create a `ProviderOrganization` association with that provider. So it fits in Providers rather than Organizations
* you don't need to introduce Provider-specific templating in `_OrganizationForm.cshtml`. That should be kept as generic as possible rather than introducing specific logic in a shared view

Reply:
> Using hidden fields to sneak important data into a POST request feels a bit hacky

I agree but this is by design when using the MVC pattern. I've removed these hidden fields and replaced that with a route parameter.

> What if we moved the endpoint to the Providers controller and put it at (for example) `providers/createorganization/{providerId}`?

Makes sense, I've moved the actions and views to that controller.

I've left `_OrganizationForm.cshtml` as it was and moved it to the shared folder because it's useful to have a shared partial view so we don't end up with duplicated code. Let me know what you think about this solution.

Reply:
Looks good, and yes I agree the shared `_OrganizationForm.cshtml` is a good idea. But at the same time we don't want it to have too many diverging code paths in a shared view, so it was good to remove that unnecessary `if`.